quizapp/quiz-loader/src/main/java/billennium/DataLoader.java

```diff
@@ -0,0 +1,84 @@
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+@RequiredArgsConstructor
+public class DataLoader implements ApplicationRunner {
+
+    private final QuizDefinitionRepository quizDefinitionRepository;
+
+    @Override
+    public void run(ApplicationArguments args) {
+        QuizDefinition quizDefinition = QuizDefinition.builder()
+                .title("JAVA")
+                .questions(createListQuestions())
+                .build();
+        quizDefinitionRepository.save(quizDefinition);
+    }
+
+    private List<Question> createListQuestions() {
+        Question question1 = Question.builder()
+                .text("Java to:")
+                .answers(createListAnswersForQuestion("Język programowania", "JVM", "C++"))
+                .build();
+        Question question2 = Question.builder()
+                .text("JVM TO")
+                .answers(createListAnswersForQuestion("JRE", "JDK", "JAVA VIRTUAL MACHINE"))
+                .build();
+        Question question3 = Question.builder()
+                .text("Co to jest konstruktor domyślny?")
+                .answers(createListAnswersForQuestion("Jest tworzony jeśli nie zdefiniowano jawnie konstruktora w klasie",
+                        "Jest tworzony tylko wtedy kiedy zostanie jawnie zdefiniowany w kodzie",
+                        "Konstruktor zdefiniowany ze słowem kluczowym default"))
+                .build();
+        Question question4 = Question.builder()
+                .text("Czy można zmienić istniejący String?")
+                .answers(createListAnswersForQuestion("Tylko za pomocą StringBuilder-a", "NIE", "Tak," +
+                        " jak każdy inny obiekt"))
+                .build();
+        Question question5 = Question.builder()
+                .text("Czy klasa może dziedziczyć kilka interfejsów?")
+                .answers(createListAnswersForQuestion("Tylko wtedy jak interfejsy roszerzają jeden wspólny interfejs",
+                        "NIE", "TAK"))
+                .build();
+
+        List<Question> listQuestion = Stream.of(question1, question2, question3, question4, question5).collect(Collectors.toList());
+        return listQuestion;
+    }
+
+    private List<Answer> createListAnswersForQuestion(String... answers) {
+        List<Answer> list = new ArrayList<>();
+        Arrays.stream(answers).forEach(a ->
+                {
+                    Answer answer = Answer.builder().text(a).correctAnswer(setCorrectAnswers(a)).build();
+                    list.add(answer);
+                }
+        );
+        return list;
+    }
+
+    boolean setCorrectAnswers(String answers) {
```

Comment:
to jest `answer` a nie `answers`

Reply:
Sposób działania tej metody jest dla mnie troche niejasny. Możesz wytłumaczyć ?

Reply:
myślałem nad sposobem dodanie wartości boolean true tam gdzie odpowiedź jest prawidłowa. Musimy to jakoś ustawić która odpowiedź jest prawidłowa.